pkg/yamlmeta/type.go

```diff
@@ -103,18 +105,48 @@ func (t *MapType) AssignTypeTo(typeable Typeable) (chk TypeCheck) {
 		chk.Violations = []string{fmt.Sprintf("Expected node at %s to be a %T, but was a %T", typeable.GetPosition().AsCompactString(), &Map{}, typeable)}
 		return
 	}
+	var foundKeys []interface{}
 	typeable.SetType(t)
 	for _, mapItem := range mapNode.Items {
 		for _, itemType := range t.Items {
 			if mapItem.Key == itemType.Key {
+				foundKeys = append(foundKeys, itemType.Key)
 				childCheck := itemType.AssignTypeTo(mapItem)
 				chk.Violations = append(chk.Violations, childCheck.Violations...)
 				break
 			}
 		}
 	}
+
+	t.applySchemaDefaults(foundKeys, chk, mapNode)
 	return
 }
+
+func (t *MapType) applySchemaDefaults(foundKeys []interface{}, chk TypeCheck, mapNode *Map) {
+	for _, item := range t.Items {
+		if contains(foundKeys, item.Key) {
+			continue
+		}
+
+		val := &MapItem{
+			Key:      item.Key,
+			Value:    item.DefaultValue,
+			Position: item.Position,
+		}
+		childCheck := item.AssignTypeTo(val)
+		chk.Violations = append(chk.Violations, childCheck.Violations...)
+		mapNode.AddValue(val) // TODO: How might we surface this error?
```

Comment:
if we are not propagating error up then we should panic. definitely do not want to swallow.

Reply:
To get an error we would need to pass a value to the method `AddValue` that is not of the type `*MapItem`. This cannot happen because we create the variable `val` 7 lines above this invocation.
Nevertheless, we can always add here a panic, not really a fan of panic in mid execution 😢  that was why I added a TODO so we can discuss the best option

Reply:
> Nevertheless, we can always add here a panic, not really a fan of panic in mid execution

`panic("Unreachable")` is a very common pattern to assert on internal constraints.

Reply:
done